Rebus.LegacyExtensions/RabbitMqMangler.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Reflection;
+
+using Rebus.Transport;
+using HermaFx;
+using Rebus.Logging;
+
+namespace Rebus
+{
+	internal class RabbitMqMangler : IDisposable
```

Comment:
Maybe I am wrong, but I do remember when we looked at this together, and what I see here does make use of too many reflection. I do not remember so many things were needed..